dali/base/dadfs.cpp

```diff
@@ -8866,18 +8867,21 @@ class CFileScanner
             CFileMatch *match = &files.item(f);
             matchingFiles.append(*LINK(match));
             ++count;
+            if (checkFileCount && (count > ITERATE_FILTEREDFILES_LIMIT))
+                throw MakeStringException(-1, "More than %d files found, please use a filter to reduce the number of files returned.",
+                    ITERATE_FILTEREDFILES_LIMIT);
```

Comment:
Also think the limit should be a default and passed in.
So that a caller could override.